### PR TITLE
#16 Split fetch projects epic into two

### LINKED ISSUE #16: Split fetch projects epic into two
* Currently `fetchAllProjects` epic does 2 tasks - fetch project list, and then for each project fetch its details. These need to be separate epics.

See the epic [here](https://github.com/srishanbhattarai/Harbor/blob/dev/src/epics/projectsEpic.js) and corresponding test [here](https://github.com/srishanbhattarai/Harbor/blob/dev/src/epics/projectsEpic.test.js).

### PR DESCRIPTION
**Pull request by mesaugat**
closes #16

* Add new `fetchProjectsDetailEpic` to fetch projects detail
* Update react-native to 0.51.0 (because the android emulator was showing some problem with 0.50.0)
* Add new tests and update previous ones
* Miner JSDoc fixes

I am not sure if it looks well and this is how you want it. It feels more clunky to me. :smile:

### CODE REVIEW

File: src/actions/projectActions.js [hunk 1]
```diff
@@ -31,20 +33,48 @@ export function fetchAllProjectsErrored(error) {
...
+  };
+}
+
+/**
+ * Error in fetching projects detail.
+ *
+ * @param {Object} error
+ * @returns {Object}
+ */
+export function fetchProjectsDetailErrored(error) {
```
Comment (srishanbhattarai) [inline src/actions/projectActions.js]: Let's change this to be FSA compilant: 
```diff
- error
+ error: true,
+ payload: error
```

Comment (mesaugat) [reply src/actions/projectActions.js]: This too?
 https://github.com/srishanbhattarai/Harbor/blob/master/src/actions/projectActions.js#L20-L31

Comment (srishanbhattarai) [reply src/actions/projectActions.js]: lol yes, that too. I don't think the errors are being used anywhere though

**Comment by srishanbhattarai** [comment]
Thanks 🙏

**Pull request merged**